### PR TITLE
Create route for IDT to get veteran by ssn

### LINKED ISSUE #10339: Add IDT route to return veteran info by file number
https://dsva.slack.com/archives/CD5DAQNCU/p1553542659001900

AC: Add IDT route to return veteran info by file number

### PR DESCRIPTION
**Pull request by hschallhorn**
Resolves #10339

### Description
Added IDT route to return veteran info by their ssn or file number

### Acceptance Criteria
- [ ] Endpoint returns basic veteran information and address
- [ ] Endpoint returns the veteran's representative information and address

### CODE REVIEW

File: app/controllers/idt/api/v1/base_controller.rb [hunk 1]
```diff
@@ -39,6 +39,10 @@ def token
     request.headers["TOKEN"]
   end
 
+  def ssn
+    request.headers["SSN"]
```
Comment (aroltsch) [inline app/controllers/idt/api/v1/base_controller.rb]: Forgot to remove?

Comment (hschallhorn) [reply app/controllers/idt/api/v1/base_controller.rb]: DANGIT

File: app/controllers/idt/api/v1/veterans_controller.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    render json: json_veteran_details
+  end
+
+  private
+
+  def veteran
+    fail Caseflow::Error::InvalidFileNumber if file_number.blank?
+
+    @veteran ||= begin
+      veteran = Veteran.find_by_file_number_or_ssn(file_number)
```
Comment (aroltsch) [inline app/controllers/idt/api/v1/veterans_controller.rb]: Do we always store veteran information in veterans table even for legacy appeals? I know we do it for AMA but not sure about legacy appeals

File: app/controllers/idt/api/v1/veterans_controller.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  end
+  # :nocov:
+
+  rescue_from ActiveRecord::RecordNotFound do |_e|
+    render(json: { message: "A veteran with that ssn or file number was not found." }, status: :not_found)
+  end
+
+  rescue_from Caseflow::Error::InvalidFileNumber do |_e|
+    render(json: { message: "Enter a file number or ssn in the 'FILENUMBER' header" }, status: :unprocessable_entity)
+  end
```
Comment (aroltsch) [inline app/controllers/idt/api/v1/veterans_controller.rb]: Let's move some of these methods into the IDT base controller and reuse them between controllers.

File: app/controllers/idt/api/v1/veterans_controller.rb (hunk 2 above)
Comment (aroltsch) [reply app/controllers/idt/api/v1/veterans_controller.rb]: Or does he only need this endpoint for ama appeals?

Comment (hschallhorn) [reply app/controllers/idt/api/v1/veterans_controller.rb]: Believe it's needed for both.

Comment (aroltsch) [reply app/controllers/idt/api/v1/veterans_controller.rb]: let's double check with Paul

File: spec/controllers/idt/api/veterans_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        before do
+          request.headers["FILENUMBER"] = file_number
+        end
+
+        it "returns the veteran's details" do
+          get :details
+          expect(response.status).to eq 200
+          response_body = JSON.parse(response.body)
+
+          expect(response_body["first_name"]).to eq veteran.first_name
```
Comment (aroltsch) [inline spec/controllers/idt/api/veterans_controller_spec.rb]: It is better to check against hard coded values. `veteran.first_name`might contain nil and we don't know about it.

File: app/controllers/idt/api/v1/base_controller.rb [hunk 5]
```diff
@@ -4,6 +4,25 @@ class Idt::Api::V1::BaseController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :validate_token
 
+  # :nocov:
+  rescue_from StandardError do |error|
```
Comment (aroltsch) [inline app/controllers/idt/api/v1/base_controller.rb]: Do you want to delete duplicate methods from other controllers?

File: app/controllers/idt/api/v1/veterans_controller.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+  def poa
+    @poa ||= begin
+      poa = bgs.fetch_poa_by_file_number(veteran[:file_number])
+      poa.merge(bgs.find_address_by_participant_id(poa[:participant_id]))
+    end
+  end
+
+  def json_veteran_details
+    veteran.merge(poa: poa)
```
Comment (aroltsch) [inline app/controllers/idt/api/v1/veterans_controller.rb]: We should probably use serializer, what do you think?

**Pull request merged**